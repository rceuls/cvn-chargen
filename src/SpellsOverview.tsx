import { getSRDLink, ILinkedItem } from './services/util';

const SpellsOverview = ({ data }: { data: ILinkedItem[] }) => {
  if (data.length) {
    return (
      <>
        {data.map((eq, ix) => (
          <div className="row" key={`id_${ix}`}>
            <div className="col-sm">
              <a href={getSRDLink(eq.link)} target="_blank" rel="noreferrer">
                {eq.label}
              </a>
            </div>
          </div>
        ))}
        <hr />
      </>
    );
  } else {
    return <></>;
  }
};

export default SpellsOverview;
