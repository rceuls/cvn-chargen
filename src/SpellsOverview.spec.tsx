import renderer from 'react-test-renderer';
import SpellsOverview from './SpellsOverview';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <SpellsOverview data={[{ label: 'Read Magic', link: 'Read_Magic' }]} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when empty', () => {
  const tree = renderer.create(<SpellsOverview data={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
