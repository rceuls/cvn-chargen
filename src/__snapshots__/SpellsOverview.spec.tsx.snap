// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <a
        href="https://oldschoolessentials.necroticgnome.com/srd/index.php/Read_Magic"
        rel="noreferrer"
        target="_blank"
      >
        Read Magic
      </a>
    </div>
  </div>,
  <hr />,
]
`;
